test/e2e/azure_logcollector.go

```diff
@@ -139,14 +141,21 @@ func collectLogsFromNode(ctx context.Context, managementClusterClient client.Cli
 
 	execToPathFn := func(outputFileName, command string, args ...string) func() error {
 		return func() error {
-			return retryWithTimeout(collectLogInterval, collectLogTimeout, func() error {
-				f, err := fileOnHost(filepath.Join(outputPath, outputFileName))
-				if err != nil {
-					return err
+			f, err := fileOnHost(filepath.Join(outputPath, outputFileName))
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			var retryError error
+			_ = wait.PollImmediate(collectLogInterval, collectLogTimeout, func() (bool, error) {
+				var execOnHostErr error
+				if execOnHostErr = execOnHost(controlPlaneEndpoint, hostname, sshPort, f, command, args...); err != nil {
+					retryError = execOnHostErr
+					return false, nil
```

Comment:
is it possible to still reuse the helper function in retry.go but modify it so it does retry after errors? That way we can reuse it later if needed